### PR TITLE
use Abseil OStringStream in WebGPU EP string concat

### PR DESCRIPTION
**Pull request by fs-eire**
### Description

use Abseil OStringStream in WebGPU EP string concat.

Allows multiple calls to shader_helper.MainFunctionBody()

### CODE REVIEW

File: onnxruntime/core/providers/webgpu/tensor/concat.cc [hunk 1]
```diff
@@ -38,48 +38,6 @@ WEBGPU_CONCAT_VERSIONED_KERNEL(4, 10)
 WEBGPU_CONCAT_VERSIONED_KERNEL(11, 12)
 WEBGPU_CONCAT_KERNEL(13)
 
-const std::string AppendCalCulateInputIndexFunction(size_t input_count) {
```
Comment (satyajandhyala) [inline onnxruntime/core/providers/webgpu/tensor/concat.cc:41]: Code from these functions are moved into the GenerateShaderCode function. Modularity is getting lost. Is this changes necessary? We could still keep these AppenXyzFunction function and use the return value to insert into additional implementation.

File: onnxruntime/core/providers/webgpu/string_utils.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+template <typename T>
+inline void OStringStreamAppendImpl(std::ostream& ss, const T& t) noexcept {
+  ss << t;
+}
+
+template <typename T, typename... Args>
+inline void OStringStreamAppendImpl(std::ostream& ss, const T& t, const Args&... args) noexcept {
+  OStringStreamAppendImpl(ss, t);
+  OStringStreamAppendImpl(ss, args...);
+}
```
Comment (skottmckay) [inline onnxruntime/core/providers/webgpu/string_utils.h:35]: Can we add something like this to decay `const char[]` to const char*? It saves significant binary size to do so.

https://github.com/microsoft/onnxruntime/blob/1bda91fc57d61367d245db808106687b43036bd1/include/onnxruntime/core/common/make_string.h#L91-L102

**Review by skottmckay — APPROVED** [review]
